clients/python/girder_client/__init__.py

```diff
@@ -520,48 +574,43 @@ def uploadFile(self, parentId, stream, name, size, parentType='item',
         if reference is not None:
             params['reference'] = reference
         obj = self.post('file', params)
-        if '_id' in obj:
-            uploadId = obj['_id']
-        else:
+        if '_id' not in obj:
             raise Exception(
                 'After creating an upload token for a new file, expected '
                 'an object with an id. Got instead: ' + json.dumps(obj))
 
-        offset = 0
-        while True:
-            data = stream.read(self.MAX_CHUNK_SIZE)
-
-            if not data:
-                break
-
-            params = {
-                'offset': offset,
-                'uploadId': uploadId
-            }
-            files = {
-                'chunk': data
-            }
-            obj = self.post('file/chunk', parameters=params, files=files)
-            offset += len(data)
+        return self._uploadContents(
+            obj, stream, size, progressCallback=progressCallback)
 
-            if '_id' not in obj:
-                raise Exception('After uploading a file chunk, did'
-                                ' not receive object with _id. Got instead: ' +
-                                json.dumps(obj))
+    def uploadFileContents(self, file_id, stream, size, reference=None):
+        """
+        Uploads the contents of an existing file..
```

Comment:
I can't tell if this is a period or ellipsis, though I'm enjoying the ambiguity!